packages/openapi-to-graphql/src/resolver_builder.ts

```diff
@@ -40,13 +49,272 @@ type GetResolverParams = {
+    )
+
+    if (typeof resolveData.usedParams === 'undefined') {
+      resolveData.usedParams = {}
+    }
+
+    if (connectOptions) {
+      resolveData.usedRequestOptions = connectOptions
+    } else {
+      resolveData.usedRequestOptions = {
+        method: resolveData.usedPayload.method
+          ? resolveData.usedPayload.method
+          : method.toUpperCase()
+      }
+    }
+
+    let value = path
+    let paramNameWithoutLocation = paramName
+    if (paramName.indexOf('.') !== -1) {
+      paramNameWithoutLocation = paramName.split('.')[1]
+    }
+
+    // /**
+    //  * see if the callback path contains constants expression
+    //  *
+    //  */
+    if (value.search(/{|}/) === -1) {
+      args[paramNameWithoutLocation] = isRuntimeExpression(value)
+        ? resolveLinkParameter(paramName, value, resolveData, root, args)
+        : value
+    } else {
+      // Replace callback expression with appropriate values
+      const cbParams = value.match(/{([^}]*)}/g)
+      pubsubLog(`Analyzing subscription path : ${cbParams.toString()}`)
+
+      cbParams.forEach(cbParam => {
+        value = value.replace(
+          cbParam,
+          resolveLinkParameter(
+            paramName,
+            cbParam.substring(1, cbParam.length - 1),
+            resolveData,
+            root,
+            args
+          )
+        )
+      })
+      args[paramNameWithoutLocation] = value
+    }
+    const topic = args[paramNameWithoutLocation] || 'test'
+    pubsubLog(`Subscring to : ${topic}`)
+    return ctx.pubsub
+      ? ctx.pubsub.asyncIterator(topic)
+      : pubsub.asyncIterator(topic)
+  }
+}
+
+/*
+ * If operationType is Subscription, creates and returns a resolver function triggered
+ * after a message has been published to the corresponding subscribe topic(s) to execute payload transformation
+ */
+export function getPublishResolver({
+  operation,
+  argsFromLink = {},
+  responseName,
+  data
+}: GetResolverParams): ResolveFunction {
+  // Return custom resolver if it is defined
+  const customResolvers = data.options.customResolvers
+  const title = operation.oas.info.title
+  const path = operation.path
+  const method = operation.method
+
+  if (
+    typeof customResolvers === 'object' &&
+    typeof customResolvers[title] === 'object' &&
+    typeof customResolvers[title][path] === 'object' &&
+    typeof customResolvers[title][path][method] === 'object' &&
+    customResolvers[title][path][method].hasOwnProperty('resolve')
+  ) {
+    translationLog(
+      `Use custom publish resolver for ${operation.operationString}`
+    )
+    const customResolver = customResolvers[title][path][method] as ResolveObject
+    if (typeof customResolver.resolve === 'function') {
+      return customResolver.resolve
+    }
+  }
+
+  return (payload: any, args, context, info = {}) => {
+    // Validate and format based on operation.responseDefinition  ...
+    const typeOfResponse = operation.responseDefinition.targetGraphQLType
```

Comment:
This function requires some improvement, but probably some discussion are needed as the formatting doesn't follow REST rules.